### DISCUSSION TITLE
[Nuxt] Global import

### DISCUSSION THREAD

**Discussion by mrleblanc101** [vueuse/vueuse#2607] [Q&A]
What is the recommanded way to import a directive globally ?

I created a plugin like so:
```js
// plugins/vueuse.ts

import { vOnClickOutside } from '@vueuse/components';

export default defineNuxtPlugin((nuxtApp) => {
    nuxtApp.vueApp.directive('on-click-outside', vOnClickOutside);
});
```

**Comment by hslee2008** [comment]
You might be interested in `@vueuse/nuxt` module which handles imports automatically.

**Comment by neilmerton** [comment]
If anyone else stumbles across this and is curious to know how to work with VueUse components within Nuxt 3, it looks like they need installing as a separate package and import explicitly.
